server/src/main/java/org/apache/iotdb/db/engine/StorageEngine.java

```diff
@@ -348,6 +352,9 @@ public void insert(InsertRowPlan insertRowPlan) throws StorageEngineException {
     // TODO monitor: update statistics
     try {
       storageGroupProcessor.insert(insertRowPlan);
+      if (config.isEnableStatMonitor()) {
```

Comment:
have you tested the performance?

I think maybe "each tsfileProcessor records the results, and then reports to the SG processor (and then reports the the Storage Engine if needed)" can reach a better performance.

The sacrifice is we can not get a real time stats.
A solution is add an JMX interface to update the value immediately, (then the tsfileProcessors report the data immediately)